pkg/blobstore/cloud/s3_copy_mutator.go

```diff
@@ -0,0 +1,34 @@
+package cloud
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/s3"
+
+	"github.com/buildbarn/bb-storage/pkg/clock"
+)
+
+type s3CopyMutator struct {
+	minRefreshAge time.Duration
+	clock         clock.Clock
+}
+
+// NewS3CopyMutator creates a BeforeCopyFunc which updates the "Used"
+// metadata field on s3 objects with the current time.
+func NewS3CopyMutator(minRefreshAge time.Duration, clock clock.Clock) BeforeCopyFunc {
```

Comment:
Could we at least add a single unit test for this type, for the success case?

Reply:
Done